cmd/installer/subcommands/daemon/run_windows.go

```diff
@@ -8,44 +8,52 @@
 package daemon
 
 import (
+	"context"
 	"github.com/DataDog/datadog-agent/cmd/installer/command"
 	"github.com/DataDog/datadog-agent/comp/core/pid"
 	"github.com/DataDog/datadog-agent/comp/updater/localapi"
 	"github.com/DataDog/datadog-agent/comp/updater/telemetry"
 	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
 	"github.com/judwhite/go-svc"
 	"go.uber.org/fx"
+	"syscall"
 )
 
 type windowsService struct {
-	global     *command.GlobalParams
-	shutdowner fx.Shutdowner
+	*fx.App
 }
 
-func runFxWrapper(global *command.GlobalParams) error {
-	return svc.Run(&windowsService{
-		global: global,
-	})
+func getFxOptions(global *command.GlobalParams) []fx.Option {
+	return []fx.Option{
+		getCommonFxOption(global),
+		fxutil.FxAgentBase(),
+		// Force the instantiation of some components
+		fx.Invoke(func(_ pid.Component) {}),
+		fx.Invoke(func(_ localapi.Component) {}),
+		fx.Invoke(func(_ telemetry.Component) {}),
+	}
 }
 
-func run(s *windowsService, shutdowner fx.Shutdowner, _ pid.Component, _ localapi.Component, _ telemetry.Component) error {
-	s.shutdowner = shutdowner
-	return nil
+func runFxWrapper(global *command.GlobalParams) error {
+	return svc.Run(&windowsService{
+		App: fx.New(getFxOptions(global)...),
+	}, syscall.SIGINT, syscall.SIGTERM)
 }
 
 func (s *windowsService) Init(_ svc.Environment) error {
 	return nil
 }
 
 func (s *windowsService) Start() error {
-	return fxutil.OneShot(
-		run,
-		getCommonFxOption(s.global),
-		fx.Supply(s),
-	)
+	// Default start timeout is 15s, which is fine for us.
+	startCtx, cancel := context.WithTimeout(context.Background(), s.StartTimeout())
+	defer cancel()
+	return s.App.Start(startCtx)
```

Comment:
❓ question

Does the App/Daemon need to trigger exit? Or does the service only ever stop when receiving a SCM signal?